tests/mocks.py

```diff
@@ -58,4 +62,79 @@ def score_h(self, rt_batch: torch.LongTensor) -> torch.FloatTensor:  # noqa: D10
         return self._generate_fake_scores(batch=rt_batch)
 
     def reset_parameters_(self) -> Model:  # noqa: D102
-        pass  # Not needed for unittest
+        self.num_backward_propagations += 1
```

Comment:
Shouldn't this be in the `post_parameter_update` hook?

Reply:
In general yes, I abuse it here to simulate the number of epochs, so I can test whether the resumed training loop model is the _best epoch model_ or not. Do you see any issues in doing it this way?

Reply:
`reset_parameters` should only be called during model instantiation, but not after each epoch, right? Or do I miss sth?

Reply:
You're absolutely right. I had mistaken it for the `post_parameter_update` and even that wouldn't reflect the number of epochs. I have changed it now to a more consistent logic :)